api/src/main/java/javax/enterprise/inject/spi/builder/AnnotatedTypeConfigurator.java

```diff
@@ -0,0 +1,432 @@
+    AnnotatedTypeConfigurator<T> removeFromMethodParameter(Method method,
+                                                           int position, Class<? extends Annotation> annotationType);
+
+    /**
+     * Add an annotation to the specified constructor. If the constructor is not
+     * already present, it will be added.
+     *
+     * @param constructor the constructor to add the annotation to
+     * @param annotation  the annotation to add
+     * @throws IllegalArgumentException if the annotation is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> addToConstructor(Constructor<T> constructor, Annotation annotation);
+
+    /**
+     * Add an annotation to the specified constructor. If the constructor is not
+     * already present, it will be added.
+     *
+     * @param constructor the constructor to add the annotation to
+     * @param annotation  the annotation to add
+     * @throws IllegalArgumentException if the annotation is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> addToConstructor(AnnotatedConstructor<T> constructor, Annotation annotation);
+
+    /**
+     * Remove an annotation from the specified constructor.
+     *
+     * @param constructor    the constructor to add the annotation to
+     * @param annotationType the annotation to add
+     * @throws IllegalArgumentException if the annotationType is null or if the
+     *                                  constructor is not currently declared on the type
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromConstructor(Constructor<T> constructor,
+                                                       Class<? extends Annotation> annotationType);
+
+    /**
+     * Remove an annotation from the specified constructor.
+     *
+     * @param constructor    the constructor to add the annotation to
+     * @param annotationType the annotation to add
+     * @throws IllegalArgumentException if the annotationType is null, if the
+     *                                  annotation does not exist on the type or if the constructor is
+     *                                  not currently declared on the type
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> removeFromConstructor(AnnotatedConstructor<T> constructor,
+                                                       Class<? extends Annotation> annotationType);
+
+    /**
+     * Add an annotation to the specified constructor parameter. If the
+     * constructor is not already present, it will be added. If the constructor
+     * parameter is not already present, it will be added.
+     *
+     * @param constructor the constructor to add the annotation to
+     * @param position    the position of the parameter to add
+     * @param annotation  the annotation to add
+     * @throws IllegalArgumentException if the annotation is null
+     * @return self
+     */
+    AnnotatedTypeConfigurator<T> addToConstructorParameter(Constructor<T> constructor,
```

Comment:
What about to add variant of this method with AnnotatedConstructor param? Imagine following use case. I already added some annotation to my **AnnotatedConstructor** (which I stored in some variable). Now I wanted to add some annotation to this constructor parameter and I cannot. I will have to get the same constructor via reflexion.